projects/saturn/src/main/java/io/fairspace/saturn/services/views/SparqlQueryService.java

```diff
@@ -376,23 +419,25 @@ private static boolean convertBooleanValue(String value) {
     }
 
     public CountDTO count(CountRequest request) {
-        var query = getQuery(request);
+        var query = getQuery(request, true);
 
         log.debug("Querying the total number of matches: \n{}", query);
 
-        var execution = QueryExecutionFactory.create(query, ds);
-        execution.setTimeout(config.countRequestTimeout);
+        try (var execution = QueryExecutionFactory.create(query, ds)) {
+            execution.setTimeout(config.countRequestTimeout);
 
-        return calculateRead(ds, () -> {
-            long count = 0;
-            try (execution) {
-                for (var it = execution.execSelect(); it.hasNext(); it.next()) {
-                    count++;
+            return calculateRead(ds, () -> {
+                var queryResult = execution.execSelect();
+                if (queryResult.hasNext()) {
+                    var row = queryResult.next();
+                    var count = row.getLiteral("count").getLong();
+                    return new CountDTO(count, false);
+                } else {
+                    return new CountDTO(-1, false);
                 }
-                return new CountDTO(count, false);
-            } catch (QueryCancelledException e) {
-                return new CountDTO(count, true);
-            }
-        });
+            });
+        } catch (QueryCancelledException e) {
+            return new CountDTO(-1, true);
```

Comment:
SparqlQueryService and JdbcQueryService return different values in this case now. Previously both returned 0, now Sparql returns -1, Jdbc 0.

Reply:
Agree, not it is different. In turn, I am not sure that 0 is the right value in case of something went wrong as potentially it is valid count value while -1 is not valid highlighting that something went wrong. Will put 0 for now

Reply:
put 0